iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -307,6 +307,318 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
+                          ArrayRef<int64_t> batchingDims,
+                          ArrayRef<int64_t> contractingDims, ShapedType shape) {
+      SmallVector<int64_t, 4> newShape(3, 1);
+      for (auto dim : batchingDims) {
+        newShape[0] *= shape.getDimSize(dim);
+      }
+      for (auto dim : contractingDims) {
+        newShape[contractingDimIndex] *= shape.getDimSize(dim);
+      }
+      auto parallelDimStartIndex = contractingDimIndex == 2
+                                       ? batchingDims.back() + 1
+                                       : contractingDims.back() + 1;
+      auto parallelDimEndIndex =
+          contractingDimIndex == 2 ? contractingDims.front() : shape.getRank();
+
+      for (int i = parallelDimStartIndex; i < parallelDimEndIndex; ++i) {
+        newShape[parallelDimIndex] *= shape.getDimSize(i);
+      }
+      return newShape;
+    };
+
+    auto lhsNewShape =
+        getNewShape(lhsContractingDimIndex, lhsParallelDimIndex,
+                    lhsBatchingDims, lhsContractingDims, lhsShapeType);
+    auto rhsNewShape =
+        getNewShape(rhsContractingDimIndex, rhsParallelDimIndex,
+                    rhsBatchingDims, rhsContractingDims, rhsShapeType);
+    SmallVector<int64_t, 4> resultNewShape = {lhsNewShape[0],
+                                              lhsNewShape[lhsParallelDimIndex],
+                                              rhsNewShape[rhsParallelDimIndex]};
+    Type dotGeneralResultType =
+        RankedTensorType::get(resultNewShape, resultType.getElementType());
+
+    Value reshapedLhs = rewriter.create<mhlo::ReshapeOp>(
+        op.getLoc(),
+        RankedTensorType::get(lhsNewShape, lhsShapeType.getElementType()),
+        op.lhs());
+    Value reshapedRhs = rewriter.create<mhlo::ReshapeOp>(
+        op.getLoc(),
+        RankedTensorType::get(rhsNewShape, rhsShapeType.getElementType()),
+        op.rhs());
+
+    auto make1DElementsAttr = [&rewriter](ArrayRef<int64_t> integers) {
+      auto type = RankedTensorType::get({static_cast<int64_t>(integers.size())},
+                                        rewriter.getIntegerType(64));
+      return DenseIntElementsAttr::get(type, integers);
+    };
+    auto dimensionNumbers = mhlo::DotDimensionNumbers::get(
+        /*lhs_batching_dimensions=*/make1DElementsAttr({0}),
+        /*rhs_batching_dimensions=*/make1DElementsAttr({0}),
+        /*lhs_contracting_dimensions=*/
+        make1DElementsAttr({lhsContractingDimIndex}),
+        /*rhs_contracting_dimensions=*/
+        make1DElementsAttr({rhsContractingDimIndex}), rewriter.getContext());
+    Value dotGeneralResult = rewriter.create<mhlo::DotGeneralOp>(
+        op.getLoc(), dotGeneralResultType, reshapedLhs, reshapedRhs,
+        dimensionNumbers,
+        op.precision_config().hasValue() ? op.precision_config().getValue()
+                                         : nullptr);
+
+    Value result = rewriter.create<mhlo::ReshapeOp>(
+        op.getLoc(),
```

Comment:
optional: I would prefer to declare a variable instead of calling `op.getLoc()` everywhere.